api/protos/bootstrap/v1/bootstrap.proto

```diff
@@ -47,6 +47,10 @@ message Upstream {
     // From https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/Duration
     // Duration only lets us represent time in 's'
     string keep_alive_time = 2;
+
+    // Timeout in milliseconds for upstream connection stream.
+    // If unset defaults to no timeout.
+    int64 timeout = 3;
```

Comment:
Can you explain how this differentiates from the the `keep_alive_time` above?

Reply:
The `keep_alive_time` tells the stream to send a ping (defaults every 5m) and would close any broken streams. The `timeout` cancels the context that each stream uses on a fixed interval forcing the connection to close and reopen. My assumption is that when the stream is cancelled it forces parts of the cache to be refreshed.